docs/source/pillars/computing_technology.md

```diff
@@ -70,6 +70,11 @@ _The tools used by researchers inside a TRE, such as programming languages, IDEs
   - Reproducibility of analyses improves auditability and accountability of how data has been used, as well as being best-practice in research.
     This may include version control software, and tools for developing and running data analysis pipelines.
   - Recommended
+* - Where possible, you should ensure the TRE automatically applies updates for software installed on TRE workspaces, with automatic updates.
+  - TRE users will benefit from the latest versions of software being installed on the TRE, to maximise the functionality at their disposal.
+    TREs that have internet access from the environment should have their software updated periodically to get the latest patches.
+    TREs that use configuration management tools can keep a record of the software installed when they were set up, including software versions.
```

Comment:
@JimMadge ^ this is more what I meant

Reply:
Do you mean,

>  TREs can keep a record of the software installed when they were set up, including software versions, via configuration management tools.

I still don't see what influence configuration management has here. You could keep records of the software whether or not you use configuration management (your package manager almost certainly does this already). I'm not sure using configuration management would make that any easier.

Reply:
It's also likely that configuration management will be higher level (eg. "this TRE needs Microsoft Office installed") rather than at the level of package versions (eg. "we have v16.74 of Microsoft Word installed").

Reply:
Configuration can be either imperative or declarative. The difference is basically the difference between telling your configuration software: "make sure that Microsoft Office is installed" or "install version v16.74 of Microsoft Word then install v16.70 of Microsoft Excel" etc.

These are both valid ways of specifying your configuration - my point is just that you can't assume that all configuration systems actually do have a config file somewhere that says "install this version of X and this version of Y", especially if you're also giving an instruction like "keep these package up-to-date".

Reply:
I'd separate configuration management from recording software versions. The former say how something is done, the latter is a form of auditing where you're identifying what software and versions are installed. If you're using config-management to build immutable images then you can get the software versions at the same time, but if you've e.g. got a large fleet of VMs that update themselves occasionally the update system will break for no reason. In this situation you might have an auditing agent that reports software versions. You might be able to remediate incorrect versions with a config management system but I think this gets into too much detail.

Reply:
I've removed this for now, I think if we want to include something about recording software versions we can do so elsewhere, and IAC is covered elsewhere in "Deployment management"